### PR TITLE
Adding CMake include guard

### PR DESCRIPTION
**Pull request by tdegeus**
Fixes https://github.com/BlueBrain/HighFive/issues/387

### CODE REVIEW

File: CMake/HighFiveTargetDeps.cmake [hunk 1]
```diff
@@ -1,79 +1,82 @@
 # Link against target system libs
 # -------------------------------
 
-# Independent target to make it possible to have new dependencies each build
-add_library(libdeps INTERFACE)
+if(NOT TARGET libdeps)
```
Comment (matz-e) [inline CMake/HighFiveTargetDeps.cmake:4]: This makes me wonder… if I declare my own target `libdeps` (the name seems pretty generic), and *then* include HighFive, wouldn't this lead to collision / failure, too?

Comment (tdegeus) [reply CMake/HighFiveTargetDeps.cmake:4]: Sure. This is the reason I have always been favour of having a makeshift namespace. Firstly, one should:
* `libdeps` -> `HighFive::dependecies`
Secondly, one could allow a finer precision:
* `HighFive::hdf5`
* `HighFive::use_Eigen`
* `HighFive::use_xtensor`
* `HighFive::use_Boost`

Comment (matz-e) [reply CMake/HighFiveTargetDeps.cmake:4]: That seems reasonable to me… also found this:

https://cmake.org/cmake/help/v3.10/command/include_guard.html?highlight=include_guard

unfortunately pretty late, only in 3.10 onwards.

Comment (tdegeus) [reply CMake/HighFiveTargetDeps.cmake:4]: `include_guard` seem nice. Just as good as the if though. Let me switch to `HighFive::dependencies` here, and then to the more refined things in separate PR.

Comment (tdegeus) [reply CMake/HighFiveTargetDeps.cmake:4]: That change is going to take a bit more work than I hoped. Personally I find HighFive's CMake support a mess... Anyway, let's do the bugfix now and improvement separately.

**Comment by tdegeus** [comment]
Ping @matz-e @alkino @ferdonline

**Review by alkino — APPROVED** [review]

**Pull request merged**